src/Stack/Build.hs

```diff
@@ -121,6 +121,11 @@ checkCabalVersion = do
             "Error: --allow-newer requires at least Cabal version 1.22, but version " ++
             versionString cabalVer ++
             " was found."
+    when (cabalVer <= mkVersion [1, 19, 2]) $ throwM $
```

Comment:
Could you explain where does 1.19.2 come from? I don't even find that version on Hackage.
BTW #4475 talks about dropping Cabal<2 though it's just a discussion yet.

Reply:
Ah right meant to link to, this comment in Cabal's code: https://github.com/haskell/cabal/blob/580fe6b6bf4e1648b2f66c1cb9da9f1f1378492c/cabal-install/Distribution/Client/Setup.hs#L592 .  I've added a comment including this link